contracts/Market.sol

```diff
@@ -1232,17 +1234,17 @@ contract Market is
    * @notice Calculates the total quantity of ERC20 tokens required to make a purchase of the specified `amount` (in
    * tonnes of carbon removals) without a transaction fee.
    * @param amount The amount of carbon removals for the purchase.
+   * @param priceMultiple The price multiple to use for the calculation.
    * @return The total quantity of ERC20 tokens required to make the purchase, excluding the fee.
    */
-  function calculateCheckoutTotalWithoutFee(uint256 amount)
-    external
-    view
-    returns (uint256)
-  {
+  function calculateCheckoutTotalWithoutFee(
+    uint256 amount,
+    uint256 priceMultiple
+  ) external view returns (uint256) {
```

Comment:
this is a new change -- I realized that externally to the contract (but nowhere internally!) this function is used to calculate the total purchase amount in ERC-20 tokens that is needed for an `approve` call.  Therefore this needs to be parameterized with the price being used for the transaction since we need to be able to calculate total purchase amount for the special order scenario (currently the only scenario in which this `withoutFee` calculation is relevant).  It's a simple enough calculation to do without the help of a contract function, because there is no fee involved, so what you're really asking is "how many NRTs am I buying and at what price?" and multiplying those two values together.  So it's almost worth removing this code from the contract entirely.  However, the reason it should stay and it's a little more complicated than NRTs * price is because of the purchasing token decimals conversion.... ultimately, the output of this function gives you the purchase total in ERC20 _in the correct decimal expression_ for the token being used.... so I think this has to stay and it has to allow specification of the price multiple directly.